### PR TITLE
Error message placement

### PR DESCRIPTION
**Pull request by Michael1104Smith**
If you made a contract change, make sure to complete the checklist below before merging it in master.

Refer to our [documentation](https://github.com/OriginProtocol/security) for more details about contract security best practices.

Contract change checklist:
  - [ ] Code reviewed by 2 reviewers. See [template](https://github.com/OriginProtocol/security/blob/master/templates/Contract-Code-Review.md) and [example](https://github.com/OriginProtocol/security/blob/master/templates/Contract-Code-Review-Example.md).
  - [ ] Unit tests pass
  - [ ] Slither tests pass with no warning
  - [ ] Echidna tests pass if PR includes changes to OUSD contract (not automated, run manually on local)

### CODE REVIEW

File: contracts/dev.env [hunk 1]
```diff
@@ -1,4 +0,0 @@
-PROVIDER_URL=[SET PROVIDER URL HERE]
```
Comment (sparrowDom) [inline contracts/dev.env:1]: Can you add this file back in?

File: dapp/src/components/buySell/SwapCurrencyPill.js [hunk 2]
```diff
@@ -282,7 +282,7 @@ const SwapCurrencyPill = ({
   const roundTo2to6Decimals = (value) => {
     return formatCurrencyMinMaxDecimals(value, {
       minDecimals: 2,
-      maxDecimals: 6,
+      maxDecimals: 20,
```
Comment (sparrowDom) [inline dapp/src/components/buySell/SwapCurrencyPill.js:285]: Function name is wrong if this change is required. Also why is this change required?

File: dapp/src/utils/contracts.js [hunk 3]
```diff
@@ -386,7 +386,7 @@ export async function setupContracts(account, library, chainId) {
       const response = await fetch(process.env.APR_ANALYTICS_ENDPOINT)
       if (response.ok) {
         const json = await response.json()
-        const apy = aprToApy(parseFloat(json.apr), 7)
+        const apy = aprToApy(parseFloat(json.apr), 30)
```
Comment (sparrowDom) [inline dapp/src/utils/contracts.js:389]: As we've discussed this in the past, there is no need to do client side APY calculation since the analytics backend already returns that info

File: dapp/styles/globals.css [hunk 4]
```diff
@@ -300,4 +300,8 @@ section.light {
   .container {
     max-width: 940px !important;
   }
+}
+
+.justify-content-right {
+  text-align: right;
```
Comment (sparrowDom) [inline dapp/styles/globals.css:306]: Bootstrap already has a `text-right` ([see api](https://getbootstrap.com/docs/4.0/utilities/text/)) you can use

File: dapp/src/components/buySell/SwapHomepage.js [hunk 5]
```diff
@@ -587,11 +596,16 @@ const SwapHomepage = ({
           </a>
           <button
             //disabled={formHasErrors || buyFormHasWarnings || !totalOUSD}
-            className={`btn-blue buy-button mt-2 mt-md-0 w-100`}
+            className={`btn-dark buy-button mt-2 mt-md-0 w-100`}
```
Comment (sparrowDom) [inline dapp/src/components/buySell/SwapHomepage.js:599]: was this change intended?

File: dapp/src/components/buySell/SwapHomepage.js [hunk 6]
```diff
@@ -587,11 +596,16 @@ const SwapHomepage = ({
...
             disabled={!selectedSwap || formHasErrors}
             onClick={onBuyNow}
           >
-            {fbt('Swap', 'Swap')}
+            {!formHasErrors
+              ? fbt('Swap', 'Swap')
+              : fbt(
+                  'Insufficient balance',
```
Comment (sparrowDom) [inline dapp/src/components/buySell/SwapHomepage.js:606]: Would be better to not hardcode the error and display the actual error here. Not sure the correct value could already be present in `formError` variable?

File: dapp/src/components/buySell/SwapCurrencyPill.js [hunk 7]
```diff
@@ -364,6 +364,21 @@ const SwapCurrencyPill = ({
         (selectedSwap.amountReceived * priceToleranceValue) / 100
       : null
 
+  const pv =
```
Comment (sparrowDom) [inline dapp/src/components/buySell/SwapCurrencyPill.js:367]: What does `pv` stand for? Can you replace with a more descriptive variable please?

File: dapp/src/components/buySell/SwapCurrencyPill.js [hunk 8]
```diff
@@ -364,6 +364,21 @@ const SwapCurrencyPill = ({
...
+    bottomItem &&
+    selectedSwap &&
+    selectedSwap.amountReceived &&
+    priceToleranceValue
+      ? -priceToleranceValue
+      : null
+  const pt =
```
Comment (sparrowDom) [inline dapp/src/components/buySell/SwapCurrencyPill.js:374]: Also more descriptive one here

**Review by sparrowDom — COMMENTED** [review]
Thanks for doing this a lot of changes in this PR 💪. Left some comments inline, could you aside from those also: 
- undo the `Flipper.json` file change
- add a description to this PR stating what are the changes / issues this PR is solving.

File: dapp/abis/Flipper.json [hunk 9]
```diff
@@ -4,100 +4,162 @@
   "sourceName": "contracts/flipper/Flipper.sol",
```
Comment (DanielVF) [inline dapp/abis/Flipper.json:4]: We should remove any changes to this file from the PR.

File: dapp/src/components/buySell/SwapCurrencyPill.js (hunk 2 above)
Comment (Michael1104Smith) [reply dapp/src/components/buySell/SwapCurrencyPill.js:285]: image

Comment (Michael1104Smith) [reply dapp/src/components/buySell/SwapCurrencyPill.js:285]: It is for "Max Including Dust #652" requirements...

File: dapp/src/components/buySell/SwapHomepage.js (hunk 5 above)
Comment (sparrowDom) [reply dapp/src/components/buySell/SwapHomepage.js:599]: From what I can tell the above comment only references the button copy. And the screenshot above is from Uniswap and not our app.

**Comment by sparrowDom** [comment]
closing due to no activity
